core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -103,7 +103,11 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
         TableScanNode left = captures.get(LEFT_TABLE_SCAN);
         TableScanNode right = captures.get(RIGHT_TABLE_SCAN);
 
-        verify(!left.isUpdateTarget() && !right.isUpdateTarget(), "Unexpected Join over for-update table scan");
+        if (left.isUpdateTarget()) {
```

Comment:
I want to write test but don't know where(which class) should put in:

- Test that simple `UPDATE` or `DELETE` query is planned with the right join, that left is the `updateTarget`.
i,e `UPDATE <table> SET shippriority = 101 WHERE custkey = (SELECT min(custkey) FROM customer`

Reply:
It feels to me that if any of the left/right is updated we should just bail out from pushdown logic.

So
```
        if (left.isUpdateTarget() && !right.isUpdateTarget()) {
            return Result.empty();
        }
```

As for testing. Please replicate the `testUpdateWithSubquery`test from `BaseFailureRecoveryTest` in `BaseConnectorTest`. This is not a FTE specific problem.